### PR TITLE
bump rust to 1.80 and rm lazy_static

### PR DESCRIPTION
**Pull request by tcoratger**
Related:
 - In Rust 1.80, `LazyCell` and `LazyLock` are stable https://blog.rust-lang.org/2024/07/25/Rust-1.80.0.html#lazycell-and-lazylock, giving us the opportunity to remove an useless dependency (`static_lazy`).
 - Similar issue is raised in reth: https://github.com/paradigmxyz/reth/issues/9799
 - `lazy_static` appears to be intended to be deprecated https://github.com/rust-lang-nursery/lazy-static.rs/issues/214

### CODE REVIEW

File: src/bin/katana_genesis.rs [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
-lazy_static! {
-    static ref GENESIS_FOLDER_PATH: PathBuf = Path::new(env!("CARGO_MANIFEST_DIR")).to_path_buf().join(".katana");
-    static ref KAKAROT_CONTRACTS_PATH: PathBuf =
-        Path::new(env!("CARGO_MANIFEST_DIR")).to_path_buf().join("lib/kakarot/build");
-    static ref COINBASE_ADDRESS: Felt = 0x12345u32.into();
-    static ref SALT: Felt = Felt::ZERO;
-}
+/// Katana genesis folder path.
+static GENESIS_FOLDER_PATH: LazyLock<PathBuf> =
+    LazyLock::new(|| Path::new(env!("CARGO_MANIFEST_DIR")).to_path_buf().join(".katana"));
```
Comment (Eikix) [inline src/bin/katana_genesis.rs:14]: What is a LazyLock? Is it the new lazy_static syntax?

**Comment by Eikix** [comment]
One question to approve this PR, are there any repos in the ecosystem that have bumped to rust 1.80? Will it break something?

Have these repos also replaced lazy_static by LazyLock?

File: src/bin/katana_genesis.rs (hunk 1 above)
Comment (tcoratger) [reply src/bin/katana_genesis.rs:14]: This is the new stable feature of Rust 1.80: https://blog.rust-lang.org/2024/07/25/Rust-1.80.0.html#lazycell-and-lazylock

**Comment by tcoratger** [comment]
> One question to approve this PR, are there any repos in the ecosystem that have bumped to rust 1.80? Will it break something?
> 
> Have these repos also replaced lazy_static by LazyLock?

Yes reth here: https://github.com/paradigmxyz/reth/issues/9799

**Comment by tcoratger** [comment]
> ?

They had `once_cell::sync::Lazy` which is equivalent but not a macro. Also I read this issue https://github.com/rust-lang-nursery/lazy-static.rs/issues/214 which seems to favor a deprecation of `lazy_static` and now that the equivalent feature is stable on std, I fear that this deprecation will accelerate so I wanted to move forward before. 

But it is continuous improvement also done to remove an additional dependency. If ever we can wait for Greg to have a third opinion on the question :)

**Comment by Eikix** [comment]
> > ?
> 
> They had `once_cell::sync::Lazy` which is equivalent but not a macro. Also I read this issue [rust-lang-nursery/lazy-static.rs#214](https://github.com/rust-lang-nursery/lazy-static.rs/issues/214) which seems to favor a deprecation of `lazy_static` and now that the equivalent feature is stable on std, I fear that this deprecation will accelerate so I wanted to move forward before.
> 
> But it is continuous improvement also done to remove an additional dependency. If ever we can wait for Greg to have a third opinion on the question :)

Last question, we've had issues with bumping rust version in the past, is this fine wrt our dependencies?

**Comment by tcoratger** [comment]
> > > ?
> > 
> > 
> > They had `once_cell::sync::Lazy` which is equivalent but not a macro. Also I read this issue [rust-lang-nursery/lazy-static.rs#214](https://github.com/rust-lang-nursery/lazy-static.rs/issues/214) which seems to favor a deprecation of `lazy_static` and now that the equivalent feature is stable on std, I fear that this deprecation will accelerate so I wanted to move forward before.
> > But it is continuous improvement also done to remove an additional dependency. If ever we can wait for Greg to have a third opinion on the question :)
> 
> Last question, we've had issues with bumping rust version in the past, is this fine wrt our dependencies?

Should be fine and as reth will bump, as always if we want to continue using reth we have to bump too... but we could put this pr on hold until we need to bump reth for ex to be 100% sure we absolutely need it.

**Review by Eikix — APPROVED** [review]
lgtm

**Comment by Eikix** [comment]
Continuous improvement doesn't wait. 👑

**Pull request merged**